libraries/BTstackLib/src/BTstackLib.cpp

```diff
@@ -396,9 +396,14 @@ BD_ADDR::BD_ADDR(void) {
 }
 
 BD_ADDR::BD_ADDR(const char * address_string, BD_ADDR_TYPE address_type) : address_type(address_type) {
-    (void) address_string;
-    // TODO: implement
-    // log_error("BD_ADDR::BD_ADDR(const char *, BD_ADDR_TYPE) not implemented yet!");
+    for (int i = 0; i < 6; i++) {
```

Comment:
I think you could replace this loop with a simple sscanf:
````
sscanf("%d:%d:%d:%d:%d:%d", address, address + 1, address + 2, address + 3, address + 4, address + 5);
````
You could also check the return value of sscanf, and if it's not 6 then set everything to 0 or `panic()` or something.

Reply:
Oh, that sounds like a better option, thank you! My C programming is very rusty since I haven't done it much since basically before 2010, the project I'm doing this for is my way of getting back into both C and embedded 😊

I'll try this out and update the PR once I've got something that feels stable!

Reply:
Hello again! I've tried out using `sscanf` now but I'm not getting it to work, when I attach a picoprobe and run a debug I can see an `isr_hardfault` in the call stack:
![bild](https://github.com/earlephilhower/arduino-pico/assets/154438/1e303967-b2ac-4ce0-ab46-3bbe5e743f3e)
I cannot understand why, I tried a simple program on replit doing just the `sscanf` and that works fine with what is supposed to be the exact same inputs and input types as well as output and output types and that works just as expected: https://replit.com/@janlindblom/tryingsscanfincpp?v=1#main.cpp

Reply:
Oh that makes sense, thank you for indulging and helping me out!
I have a new version of the implementation now that I just tried and that seems like it's working, it would seem I had to use `%hhx` to find the hex representation of the byte. I'll make or update an example to show constructor in use too but it seems like it's working now!